source/index.js

```diff
@@ -184,20 +184,48 @@ module.exports = async (input = 'patch', options) => {
 			enabled: () => options.yarn === true,
 			skip: () => {
 				if (options.preview) {
-					return `[Preview] Command not executed: yarn version --new-version ${input}.`;
+					let previewText = `[Preview] Command not executed: yarn version --new-version ${input}`;
+
+					if (options.message) {
+						previewText += ` -m '${options.message.replace(/%s/g, input)}'`;
+					}
+
+					return `${previewText}.`;
 				}
 			},
-			task: () => exec('yarn', ['version', '--new-version', input])
+			task: () => {
+				const args = ['version', '--new-version', input];
+
+				if (options.message) {
+					args.push('-m', options.message);
```

Comment:
for yarn, i dont think think the -m flag can set the version. If you follow the yarn docs [here](https://classic.yarnpkg.com/en/docs/cli/version/#toc-git-tags), you'll notice that yarn needs a separate `yarn config set version-git-message "v%s"` to actually set the custom commit message.

`yarn` with -m does not work for me on windows 10 with yarn v2.4.0

Reply:
I'm aware of this but thought that `config set version-git-message "v%s"` would change the actual version tag, not the git **commit** message, but I guess it's for the commit message. Also, Yarn 2 doesn't support the `-m` flag, no, Yarn 1 does, but undocumented. Also Yarn 2 doesn't support `config set version-git-message`, so I don't really know how Yarn 2 can even be supported here. This why I just chose to use the `-m` flag, especially because it's consistent with the npm way. I mean, sure, I can change it to `yarn config set version-git-message "v%s"` if you want, but the outcome would be the same.

Reply:
I just checked out the Yarn 2 docs, and you're right, the version commit config apparently is missing from it. I have opened a [discussion](https://github.com/yarnpkg/berry/discussions/2438) on the berry repo asking if the community has any idea about how it got missed.

Also a heads up, for me on Windows 10 with classic yarn, the `-m` flag does not work. I need to use `config set version-git-message "v%s"` to actually set a custom message even for Yarn 1. Is this specific to Windows or is it the same for other UNIX distributions? Did you manually test on any system?

Reply:
Whoops, I just looked at https://github.com/yarnpkg/yarn/blob/master/src/cli/commands/version.js again, Yarn 1 does support `--message` but not the alias `-m`, seems like I misread that. But `--message` should work, I tested that out.

Thanks for opening the discussion. @arcanis already answered there and said the `version` command in Yarn 2 doesn't make a commit at all, which is what I already expected unfortunately...

Honestly, I don't know how to continue here. I will change the `-m` flag to `--message` here but other than that, it seems like we have to detect the Yarn version, and if it is version 2, create a custom git commit?

We could also just add the info that `yarn version` doesn't create a commit in version 2 to the docs, since it was always like that anyways, right?

Reply:
I agree, we can make this compatible with classic yarn but if 2 does not create a commit at all, it needs to considered separately. You can change the -m to --message and that should be enough in this PR. Any thoughts, @sindresorhus ?

Reply:
Alright, I pushed the change. Should I change it to `--message` for the npm command as well, just to make it more consistent?

Reply:
> and that should be enough in this PR

I'm not sure, I've added no tests for this whatsoever but I also don't really know how to do code them currently, as I said here: https://github.com/sindresorhus/np/pull/597#issuecomment-771411856

Reply:
> Should I change it to --message for the npm command as well, just to make it more consistent?

Yes. In general, I always prefer long flags whenever possible.

Reply:
> Yes. In general, I always prefer long flags whenever possible.

Alright, I replaced it there too now.